app/code/scout/twitter.ts

```diff
@@ -32,7 +32,48 @@ export async function getUserData(username: string) {
   try {
     const response = await axios.get(getUserEndpoint, config);
     return response.data;
-  } catch (error:any) {
+  } catch (error: any) {
     console.error(`Error fetching user data: ${error.message}`);
   }
 }
+
+export async function fetchTwitterData(twitterUsername: string): Promise<any> {
+  const options = {
+    headers: {
+      Authorization: `Bearer ${TWITTER_BEARER_TOKEN}`
+    }
+  };
+  const getUserEndpoint =
+    "https://api.twitter.com/2/users/by?usernames=" + twitterUsername;
+  const getUserResponse1 = await fetch(getUserEndpoint, options);
+  const getUserResponseJson = await getUserResponse1.json();
+  const userId = getUserResponseJson.data[0].id;
+  const getTweetsEndpoint = `https://api.twitter.com/2/users/${userId}/tweets?tweet.fields=created_at&max_results=5`;
+  console.log("\n= = = = = SENDING TWITTER REQUEST = = = = = \n");
+  const getTweetsResponse = await fetch(getTweetsEndpoint, options);
+  const getTweetsData = await getTweetsResponse.json();
+  const lastTweet = getTweetsData.data[getTweetsData.data.length - 1];
+  console.log("Last Tweet: ", lastTweet);
+  console.log("response from twitter:\n" + JSON.stringify(getTweetsData));
+  const twitterUnixTimestamp = Date.parse(lastTweet.created_at) / 1000;
+  const userData = await getUserData(twitterUsername);
+  const user = userData.data[0];
+  const authorName = user.name;
+  const description = user.description;
+  const profileImageUrl = user.profile_image_url;
+  const followersCount = user.public_metrics.followers_count;
+  console.log("Author name:", authorName);
+  console.log("Description:", description);
+  console.log("Profile image URL:", profileImageUrl);
+  console.log("Followers:", followersCount);
+
+  return {
```

Comment:
one main problem here:

we don't want to be replacing the main "scrapeData" summary, content type, image, etc. with information from twitter.

the ONLY data we want from twitter is the author name, time updated, time updated source, and follower count as likes.  Everything else should remain unchanged.

1. Update the twitter.fetchTwitterData to return only the data we need from this.
2. Add a check that the bearer token exists and if not, log a warning and return null
3. wrap it in a try-catch.  If the request to twitter fails log a warning and return null

4. the scrapePage file should be responsible for calling the twitter.fetchTwitterData, and if there is valid data updating the ScrapedInfo object it will return

Reply:
Completed this.